src/Middleware/CasProxyTicketSessionMiddleware.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace OpenEuropa\EPoetry\Middleware;
+
+use Http\Promise\Promise;
+use Phpro\SoapClient\Middleware\Middleware;
+use Phpro\SoapClient\Middleware\MiddlewareInterface;
+use Psr\Http\Message\RequestInterface;
+use Symfony\Component\HttpFoundation\Session\Session;
+
+class CasProxyTicketSessionMiddleware extends Middleware implements MiddlewareInterface
+{
+    private const PGT_ATTRIBUTE = 'cas_pgt';
+
+    /**
+     * The user session.
+     *
+     * @var Symfony\Component\HttpFoundation\Session\Session
+     */
+    protected $session;
+
+    /**
+     * CasProxyTicketMiddleware constructor.
+     *
+     * @param session $session
+     *   The user session
+     */
+    public function __construct(Session $session)
+    {
+        $this->session = $session;
+    }
+
+    public function beforeRequest(callable $handler, RequestInterface $request): Promise
+    {
+        $proxyTicket = $this->getProxyTicket();
+        if (empty($proxyTicket)) {
+            throw new \Exception('[epoetry] session has no proxy ticket.');
```

Comment:
Should we throw an exception here or just log that no PROXY ticket has been found? We could inject a PSR3 logger interface in the constructor and use `\Psr\Log\NullLogger` for tests.

Also important, we need to make sure we align with work done by @imanoleguskiza on OPENEUROPA-1551, but that does not block this ticket, I think.

Reply:
For me, the proxy ticket is mandatory. Not having it should throw an error and prevent the request from being sent (it would return error anyway).

Reply:
I don't see it as mandatory, we don't need when running tests for example, and we might not need it in certain configurations/scenarios, we need when using the library on production, that's for sure.

Reply:
We need integration with logger and a strategy for validation.
For now I propose to leave the exception.